src/store/state/state-model.ts

```diff
@@ -157,7 +157,7 @@ export const stateModel: StateModelIF = {
       }
     }
   },
-  originalSnapshot: [],
+  originalSnapshot: null,
```

Comment:
Needs to be null because this is now an object. This affects 2 things:
1. need to use optional chaining when de-referencing a property (eg, `originalSnapshot?.myProperty`)
2. the unit tests need to assign an object to this

^^ I have made these changes to current code. FYI for future code.